### ISSUE TITLE
Output Transformed Query Document to a Specified Directory

### ISSUE THREAD

**Issue by JovannyEspinal** [apollographql/apollo-ios#929]
Opening an iOS issue to its Android counterpart.

> I'm dealing with an API that has persisted operations, but doesn't support Apollo's APQ. I have to manually upload a manifest of the queries. If I do this using the human-friendly .graphql queries, they're missing the __typenames needed, and the operation hashes don't match.
I'd like a way to get the __typename'd query documents at build time. They're present in the generated query classes.

A user of the Apollo-Android sdk had a nearly identical issue that was fixed here: https://github.com/apollographql/apollo-android/pull/1599

If possible, it would be great to have this feature on the iOS sdk as well!

**Comment by designatednerd** [comment]
Once you've generated the code, you should be able to instantiate queries and access this from from the `queryDocument` property.

If you use the `--operationIdsPath`, it also puts out a JSON file that has a `source` property for each operation. That property allows you to access the raw source of an operation (though it's going to be slightly different than the `queryDocument` if you're using fragments, since it does not include the fragments referenced in the operation).

Does this cover what you're looking for?

**Comment by JovannyEspinal** [comment]
Unfortunately not.

The problem is that we need to send queries that have `__typename` _and_ bundled fragments to our backend to do some syncing that we do to support our own persisted queries...similar to the Android fix that I linked above. Do we have anything close to that on iOS?

**Comment by designatednerd** [comment]
The `queryDocument` property includes both all that `__typename` goo and the bundled fragments. [Here's an example of that in action](https://github.com/apollographql/apollo-ios/blob/94101d2ad8bc8edb369d72b2df75cb369a286697/Tests/StarWarsAPI/API.swift#L1010). 

Looking at the fix on Android, it looks like they've added an option to Gradle to handle that output. Unfortunately, it's not quite as simple right now to output that into a separate file for iOS as it is on Android (looooooong story but basically this is part of why [I'm gonna rewrite codegen in Swift](https://github.com/apollographql/apollo-ios/projects/2)). 

Does the updated explanation about `queryDocument` address your concern? If not, I can add this as an enhancement to the Swift codegen project, though I can't make any guarantees about timeline, unfortunately.

**Comment by designatednerd** [comment]
I'm gonna close this issue out, but track this enhancement for Swift Codegen through #1083 just to have everything in one place. Please follow that issue if you're interested in how that's all going. 

TL;DR it'll happen, but probably not in v0.0.1alpha.
